Symfony/CS/Console/Command/SelfUpdateCommand.php

```diff
@@ -95,4 +103,43 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
     }
+
+    private function checkIfVersionFileExists($major, $minor, $patch)
+    {
+        $url = $this->buildVersionFileUrl($major, $minor, $patch);
+        $headers = get_headers($url);
+
+        return stripos($headers[0], '200 OK') ? true : false;
+    }
+
+    private function findBestVersion($major, $minor, $patch)
+    {
+        if ($this->checkIfVersionFileExists($major, $minor, $patch + 1)) {
+            return $this->findBestVersion($major, $minor, $patch + 1);
+        }
+
+        if ($this->checkIfVersionFileExists($major, $minor + 1, 0)) {
+            return $this->findBestVersion($major, $minor + 1, 0);
+        }
+
+        if ($this->checkIfVersionFileExists($major + 1, 0, 0)) {
+            return $this->findBestVersion($major + 1, 0, 0);
+        }
+
+        return array($major, $minor, $patch);
+    }
+
+    private function buildVersionFileUrl($major, $minor, $patch)
+    {
+        return sprintf('http://get.sensiolabs.org/php-cs-fixer-v%s.phar', $this->buildVersionString($major, $minor, $patch));
+    }
+
+    private function buildVersionString($major, $minor, $patch)
+    {
+        if (0 === $patch && $major < 2) {
+            return sprintf('%d.%d', $major, $minor);
```

Comment:
actually, we could just relesae 1.12.0 instead of 1.12 and dont have that exception, as it won't affect versions 1.11(.0), 1.10(.0) or lower (as the patch will be release for 1.11.4+)